Support for tea, dose, RPM, UI overhaul, fixes, expo 53, dependency updates

CrazyCoder opened the pull request:
- support for tea recipe cards
- automatically fix invalid tea recipes that fail to steep (have >90 ml pours)
- proper custom dose support
- grinder RPM support
- disable grinder (not perfect, needs loading another recipe with grinder first to work)
- fix loading shared recipes (get XID and correct pour pattern, grind RPM, cup type)
- major UI overhaul
- code cleanup and formatting
- expo 53
- updated dependencies, removed sentry

CrazyCoder committed:
- support for tea recipe cards
- automatically fix invalid tea recipes that fail to steep (have >90 ml pours)
- proper custom dose support
- grinder RPM support
- disable grinder (not perfect, needs loading another recipe with grinder first to work)
- fix loading shared recipes (get XID and correct pour pattern, grind RPM, cup type)
- major UI overhaul
- code cleanup and formatting
- expo 53
- updated dependencies, removed sentry

CrazyCoder committed:
refactor edit recipe logic to improve maintainability and reduce code duplication

CrazyCoder committed:
replace simple getters / setters with direct field access

CrazyCoder committed:
add tooltips for cup type and grinder on/off controls

CrazyCoder committed:
update tamagui

CrazyCoder committed:
helpers for debugging / testing

CrazyCoder committed:
different colors for Tea cards, ellipsize long titles

CrazyCoder committed:
fix typo

CrazyCoder committed:
RPM cannot be < 60

CrazyCoder committed:
don't try to disable grinder for tea

CrazyCoder committed:
fix PortalProvider

CrazyCoder committed:
- workaround for non-original NFC cards that don't support "read multiple blocks" command
- proper cup types
- prepare for backup / restore support
- code cleanup

CrazyCoder committed:
update dependencies

CrazyCoder committed:
hopefully fixed the PortalProvider

CrazyCoder committed:
- support for tea cups / steeps
- support for longer pauses (360s)

CrazyCoder committed:
tsconfig update

CrazyCoder committed:
fix for steeps/cups parsing and logging

CrazyCoder committed:
cache the Recipe object with useMemo

CrazyCoder committed:
disable ShareIntentProvider debug

CrazyCoder committed:
copy previous pour values when adding a new one

CrazyCoder committed:
- update tooltip descriptions
- don't allow using more tea steeps than pours added

CrazyCoder committed:
bump version to 2.1.0

CrazyCoder committed:
hotfix: reorder cups

CrazyCoder committed:
hotfix: reorder cups (formatting)

CrazyCoder committed:
make +/- buttons more responsive

CrazyCoder committed:
bump dependencies

CrazyCoder committed:
make tea and coffee recipes separate, don't allow changing the cup type to tea and vice versa

CrazyCoder committed:
- automatic fetching of recipe title by XID
- remove the user configurable number of tea cups to avoid confusion, number of pours will be used instead
- flexible layout for text input fields
- better volume tooltip

CrazyCoder committed:
properly handle pause time == 256 since the latest firmware supports pauses up to 360 seconds

CrazyCoder committed:
restore recipes from NFC / by XID / by Share Link

CrazyCoder committed:
pass the current app version so that the API doesn't fail with long pause recipes

CrazyCoder committed:
get rid of prefixArray/suffixArray as backup field replaces it

CrazyCoder committed:
store a copy of recipe data downloaded from the online database to allow restoring from it while offline

CrazyCoder committed:
bump dependencies

CrazyCoder committed:
stick Restore / Save buttons to the bottom of the screen

CrazyCoder committed:
bump version

CrazyCoder committed:
bump deps

CrazyCoder committed:
Prepare for app store releases under XBRecipeWriter+ name.

czuryk committed:
Prepare for app store releases under XBRecipeWriter+ name.

czuryk committed:
Merge remote-tracking branch 'refs/remotes/origin/main'

terminaldisclaimer commented:
Is there a way I can transfer the App Store name? I’m happy to do it. Or I can even remove it. There is no pride of authorship here, I’d just prefer for it to not be confusing. And I likely don’t have time to continue this project

CrazyCoder committed:
bump deps

CrazyCoder committed:
fix Import dialog closing and invisible buttons with large fonts

CrazyCoder committed:
limit max dialog height to 90% of the screen and enable animation

CrazyCoder committed:
bump deps, remove unused components

CrazyCoder committed:
remove unused component patch

CrazyCoder committed:
single snap point for dialogs

CrazyCoder committed:
formatting

CrazyCoder committed:
major UI and performance improvements:
- don't serialize/deserialize the Recipe object on every change
- properly disable scrolling when using sliders (now sliders work fine on iOS)
- prettify volume display and make volume update almost real-time when dragging sliders

CrazyCoder committed:
support for long pressing +/- buttons to quickly change values; the longer you hold, the faster it will change

CrazyCoder committed:
get rid of unnecessary setRecipe/getRecipe

CrazyCoder committed:
- volume UI updates are instant now (even when dragging the slider)
- fix for agitation pattern setting not saving

CrazyCoder committed:
small performance tweak

CrazyCoder committed:
faster start of value changes on long press

CrazyCoder committed:
unused code cleanup
